### ISSUE TITLE
Date Signup Form Field validation not running

### ISSUE THREAD

**Issue by mewalsh** [aws-amplify/amplify-flutter#3757]
### Description

When attempting to add custom validation to a signup form, validation never runs for the birthdate field.  Other fields seem to run the validation as expected.  

With the example below, we always get the `FAIL` error message, however our custom `birthdate` validator is never invoked.

```dart
SignUpForm.custom(fields: [
        SignUpFormField.givenName(
          required: true,
          validator: (value) {
            print("Running Name Validation");
            return "FAIL";
          },
        ),
        SignUpFormField.familyName(required: true),
        SignUpFormField.birthdate(
          required: true,
          validator: (value) {
            print("Running Age Validation");
            return 'ALWAYS FAIL TO TEST';
          },
        ),
        SignUpFormField.username(),
        SignUpFormField.password(),
        SignUpFormField.passwordConfirmation()
      ])
```

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [X] Auth
- [X] Authenticator
- [ ] DataStore
- [ ] Notifications (Push)
- [ ] Storage

### Screenshots

![image](https://github.com/aws-amplify/amplify-flutter/assets/44404606/808a0bfe-7581-4944-8aca-14ad8539e1f4)

### Platforms

- [X] iOS
- [ ] Android
- [ ] Web
- [ ] macOS
- [ ] Windows
- [ ] Linux

### Flutter Version

3.13.4

### Amplify Flutter Version

1.4.1

### Deployment Method

Amplify CLI

**Comment by Jordan-Nelson** [comment]
Hello @mewalsh - I was able to reproduce this. It looks like this is a bug impacting date fields. I will look into a fix for this. Thanks for opening the issue.

**Comment by Jordan-Nelson** [comment]
@mewalsh - I have opened a fix for this

**Comment by Jordan-Nelson** [comment]
The PR above has been merged and should be included in the next release.

**Comment by Jordan-Nelson** [comment]
Hi @mewalsh the fix was included in version 1.4.2. Thanks again for opening this issue.
